### PR TITLE
Add cleanFinalizer controller 

### PR DESCRIPTION
**Pull request by cwen0**
Signed-off-by: Cwen Yin <cwenyin0@gmail.com>

### What problem does this PR solve?

.  

After this pr, we could ensure that "recover" could be at least invoked at once even with the annotation `chaos-mesh.chaos-mesh.org/cleanFinalizer:forced`

### What's changed and how it works?

. 

Add cleanFinalizer controller after `record` controller. 

### Related changes

- [ ] This change also requires further updates to the [website](https://github.com/chaos-mesh/website) (e.g. docs)
- [ ] This change also requires further updates to the `UI interface`
- Need to **cheery-pick to release branches**
  - [x] release-2.3
  - [x] release-2.2

### Checklist

CHANGELOG

- [ ] I have updated the `CHANGELOG.md`
- [ ] I have labeled this PR with "no-need-update-changelog"

Tests

- [ ] Unit test
- [ ] E2E test
- [ ] No code
- [ ] Manual test (add steps below)

Side effects

- [ ] **Breaking backward compatibility**

### DCO

If you find the DCO check fails, please run commands like below (Depends on the actual situations. For example, if the failed commit isn't the most recent) to fix it:

```shell
git commit --amend --signoff
git push --force
```

### CODE REVIEW

File: controllers/common/finalizers/controller.go [hunk 1]
```diff
@@ -51,8 +51,74 @@ type Reconciler struct {
...
+	if !obj.IsDeleted() {
+		if !ContainsFinalizer(obj.(metav1.Object), RecordFinalizer) {
+			r.Recorder.Event(obj, recorder.FinalizerInited{})
+			shouldUpdate = true
+			finalizers = append(obj.GetFinalizers(), RecordFinalizer)
+		}
+	}
+
+	if shouldUpdate {
+		updateError := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
```
Comment (Andrewmatilde) [inline controllers/common/finalizers/controller.go:79]: There are two duplicate parts of update code in two Reconciler. Make a function here is better.

**Review by Andrewmatilde — COMMENTED** [review]
Plz update README.md in finalizer fold. Rest LGTM.

File: controllers/common/finalizers/controller.go (hunk 1 above)
Comment (cwen0) [reply controllers/common/finalizers/controller.go]: @Andrewmatilde Fixed! PTAL again!

File: controllers/common/finalizers/controller.go [hunk 2]
```diff
@@ -51,8 +51,48 @@ type Reconciler struct {
...
+	if !obj.IsDeleted() {
+		if !ContainsFinalizer(obj.(metav1.Object), RecordFinalizer) {
+			r.Recorder.Event(obj, recorder.FinalizerInited{})
+			shouldUpdate = true
+			finalizers = append(obj.GetFinalizers(), RecordFinalizer)
+		}
+	}
+
+	if shouldUpdate {
+		return updateFinalizer(r.ReconcilerMeta, obj, req, finalizers)
```
Comment (Andrewmatilde) [inline controllers/common/finalizers/controller.go:84]: You can just move  `updateFinalizer(r.ReconcilerMeta, obj, req, finalizers)`  after line 79 & remove shouldUpdate.

**Review by Andrewmatilde — APPROVED** [review]
Rest LGTM

File: controllers/common/finalizers/controller.go (hunk 2 above)
Comment (cwen0) [reply controllers/common/finalizers/controller.go:84]: Yes, you are right.

Comment (cwen0) [reply controllers/common/finalizers/controller.go]: Fixed

**Review by Andrewmatilde — APPROVED** [review]
LGTM

**Review by FingerLeader — APPROVED** [review]
LGTM

**Pull request merged**